client/store/cart.js

```diff
@@ -0,0 +1,140 @@
+import axios from "axios";
+
+// ACTION TYPES
+const SET_CART = "SET_CART";
+const UPDATE_CART = "UPDATE_CART";
+
+// ACTION CREATORS
+const setCart = (cart) => {
+  return {
+    type: SET_CART,
+    cart,
+  };
+};
+
+const _updateCart = (updatedCart) => {
+  return {
+    type: UPDATE_CART,
+    updatedCart,
+  };
+};
+
+// THUNKS
+const TOKEN = "token";
+export const fetchCart = () => {
+  return async (dispatch) => {
+    try {
+      const token = window.localStorage.getItem(TOKEN);
+      
+      if(token) {
+        const { data: cart } = await axios.get("/api/cart", {
+          headers: {
+            authorization: token,
+          },
+        });
+        console.log("🍇", cart);
+        dispatch(setCart(cart));
+      } else {
+        const { data: cart } = await axios.get("/api/cart");
+        console.log("🥬", cart);
+        dispatch(setCart(cart));
+      }
+    } catch (error) {
+      console.error(error);
+    }
+  };
+};
+
+export const addToCartThunk = (product) => {
+  return async (dispatch) => {
+    try {
+      const token = window.localStorage.getItem(TOKEN);
+      
+      if(token) {
+        const { data: addToCart } = await axios.post(
+          "/api/cart/addToCart",
+          product,
+          {
+            headers: {
+              authorization: token,
+            },
+          }
+        );
+        console.log("🍋", addToCart);
```

Comment:
All console logs for debugging and commented code that isn't a TODO should be removed before going into the main branch